src/workerd/api/encoding.c++

```diff
@@ -350,7 +351,30 @@ kj::Maybe<jsg::JsString> IcuDecoder::decode(
           omitInitialBom = data[0] == 0xfeff;
           bomSeen = true;
         }
-        return js.str(data.slice(omitInitialBom ? 1 : 0, data.size()));
+
```

Comment:
hmm.. if these didn't fail before and we'd just return something invalid, then this *might* need to be gated by the `pedanticWpt` flag, unfortunately. But just to confirm: what is the original behavior here? would it just return a corrupted response? Would it error?

Reply:
We can't simply add a pedanticWpt flag check to here due to circular dependencies...